app/models/merchant.rb

```diff
@@ -1,5 +1,40 @@
 class Merchant < ApplicationRecord
   has_many :items
+  has_many :invoice_items, through: :items
+  has_many :invoices, through: :invoice_items
 
   validates :name, presence: true
+
+  before_validation :default_status
+
+  def toggle_disabled
+    update(disabled: !disabled)
+  end
+
+  def default_status
+    self.disabled = true if self.disabled.nil?
+  end
+
+  def self.top_revenue
+    joins(invoices: :transactions)
+      .where(transactions: { result: 1 })
+      .group('merchants.id')
+      .order(Arel.sql('SUM(invoice_items.quantity * invoice_items.unit_price) DESC'))
+      .limit(5)
+  end
+
+  def revenue
+    invoices
+      .joins(:transactions)
+      .where(transactions: { result: 1 })
+      .sum('invoice_items.quantity * invoice_items.unit_price')
+  end
+
+  def top_date
+    invoices
+      .joins(:invoice_items)
+      .group('DATE(invoices.created_at)')
+      .sum('invoice_items.quantity * invoice_items.unit_price')
+      .max_by { |_, revenue| revenue }&.first
```

Comment:
I believe max_by is a ruby method, so we may want to refactor this later.

Reply:
it 100% is

Reply:
I refactored it to use the Arel.sql